File: Backend_compiler.java
import java.util.*;

public class BackendCompiler {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter TAC (t1=a+b): ");
        String input = sc.nextLine();

        char a = input.charAt(3);
        char op = input.charAt(4);
        char b = input.charAt(5);

        System.out.println("\nAssembly Code:");

        System.out.println("MOV " + a + ", R0");

        switch(op) {
            case '+':
                System.out.println("ADD " + b + ", R0");
                break;
            case '-':
                System.out.println("SUB " + b + ", R0");
                break;
            case '*':
                System.out.println("MUL " + b + ", R0");
                break;
            case '/':
                System.out.println("DIV " + b + ", R0");
                break;
        }

        System.out.println("MOV R0, RESULT");

        sc.close();
    }
}

File: Backend_compiler.c
#include<stdio.h>

int main() {
    char op;
    char a, b, c;

    printf("Enter TAC (example: t1=a+b): ");
    scanf("%*[^=]=%c%c%c", &a, &op, &b);

    printf("\nAssembly Code:\n");

    printf("MOV %c, R0\n", a);

    switch(op) {
        case '+':
            printf("ADD %c, R0\n", b);
            break;
        case '-':
            printf("SUB %c, R0\n", b);
            break;
        case '*':
            printf("MUL %c, R0\n", b);
            break;
        case '/':
            printf("DIV %c, R0\n", b);
            break;
    }

    printf("MOV R0, RESULT\n");

    return 0;
}